src/util.h

```diff
@@ -1,30 +1,88 @@
 #include <fstream>
+#include <string.h>
+#include "solver.h"
 
-void file_to_grids(std::string filename, std::vector<Grid> &grids){
+
+struct StartupOptions
+{
+    std::string input_file = "simple-500";
+    Solver *solver = new BFS();
```

Comment:
I think it might be better to have this without new. Then in the other file get &solver.
This way I think we can get rid of mem leaks (though probably should double check with valgrind)

Reply:
Thinking about this for the long term.
Maybe we should be a map:
{
"DFS": DFS(),
"BFS": BFS(),


}

If the option is "-solver DFS", we grab and look up in the map DFS, then this Solvers would just be a list with just DFS in it. But the default, since we need to run timing on all of them later. We If no argument is specified, we grab all the values and end up with a list like [DFS(), BFS().... etc] in which we loop over. What are your thoughts?